src/main/java/core/basesyntax/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.StorageImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.Balance;
+import core.basesyntax.operation.Purchase;
+import core.basesyntax.operation.Return;
+import core.basesyntax.operation.Supply;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationService> strategyMap;
+
+    public OperationStrategy(StorageImpl storage) {
+        this.strategyMap = createStrategyMap(storage);
+    }
+
+    private Map<FruitTransaction.Operation, OperationService> createStrategyMap(StorageImpl data) {
+        Map<FruitTransaction.Operation, OperationService> map = new HashMap<>();
+        map.put(FruitTransaction.Operation.BALANCE, new Balance(data));
```

Comment:
```suggestion
        map.put(FruitTransaction.Operation.BALANCE, new BalanceService(data));
```
let's rename all classes that extends OperationService in this way.